components/library/RichTextEditor/Preview/RichTextEditor.tsx

```diff
@@ -0,0 +1,135 @@
+import { Box, Icon, InlineGrid, Text } from "@shopify/polaris";
+import { AlertCircleIcon } from "@shopify/polaris-icons";
+import { useEffect, useState } from "react";
+import type { ReactQuillProps } from "react-quill";
+import "react-quill/dist/quill.snow.css";
+import { ClientOnly } from "remix-utils/client-only";
+
+type ReducedQuillProps = Omit<ReactQuillProps, 'readOnly' | 'className' | 'theme' | 'children' | 'style' | 'tabIndex' | 'scrollingContainer'>
+
+const defaultModuleOptions = {
+    toolbar: [
+        ["bold", "italic", "underline", "blockquote"],
+        [
+            { list: "ordered" },
+            { list: "bullet" },
+            { indent: "-1" },
+            { indent: "+1" },
+        ],
+        ["link"],
+        ["clean"],
+    ],
+    clipboard: {
+        matchVisual: false // Needed to avoid this glitch: https://github.com/slab/quill/issues/2905
+    }
+};
+
+export function RichTextEditor({
+    bounds,
+    defaultValue,
+    formats,
+    id,
+    modules,
+    onBlur,
+    onChange,
+    onChangeSelection,
+    onFocus,
```

Comment:
For the props that aren't used other than passing to the `Quill` component (which looks like most of them), can you put them in a `...rest` prop and then pass them to the `Quill` component to make it a little cleaner?

Looks like these are the only props you'd really need outside of the `...rest`:
`export function RichTextEditor({ modules, label, labelHidden = false, disabled, error, ...rest }) {...}`

Then you can just render the `Quill` component like this:
`<Quill
        className={className}
        modules={mergedModuleOptions}
        theme='snow'
        readOnly={disabled}
        {...rest}
      />`

Reply:
If we make this change then I think you won't get any autocomplete/suggestions for component properties in your IDE.
![image](https://github.com/user-attachments/assets/c0903998-a9c2-4a1c-b6dd-b3d2319e98e7)

I think it's a lot handier for the developer to see all the props they can pass or even just easily see which ones they have/haven't done yet.
![image](https://github.com/user-attachments/assets/3666fe2a-0c87-41b3-baf2-c37951afaccf)

Reply:
Yeah that’s a good point, I’m fine with leaving those in there